include/text.php

```diff
@@ -186,42 +184,33 @@ function get_cats_and_terms($item)
 {
 	$categories = [];
 	$folders = [];
-
-	$matches = [];
 	$first = true;
-	$cnt = preg_match_all('/<(.*?)>/', $item['file'], $matches, PREG_SET_ORDER);
-	if ($cnt) {
-		foreach ($matches as $mtch) {
-			$categories[] = [
-				'name' => XML::escape(FileTag::decode($mtch[1])),
-				'url' =>  "#",
-				'removeurl' => ((local_user() == $item['uid'])?'filerm/' . $item['id'] . '?f=&cat=' . XML::escape(FileTag::decode($mtch[1])):""),
-				'first' => $first,
-				'last' => false
-			];
-			$first = false;
-		}
+
+	foreach (FileTag::fileToArray($item['file'], 'category') as $savedFolderName) {
+		$categories[] = [
+			'name' => $savedFolderName,
+			'url' => "#",
+			'removeurl' => ((local_user() == $item['uid']) ? 'filerm/' . $item['id'] . '?f=&cat=' . rawurlencode($savedFolderName) : ""),
+			'first' => $first,
+			'last' => false
+		];
+		$first = false;
 	}
 
 	if (count($categories)) {
 		$categories[count($categories) - 1]['last'] = true;
 	}
 
 	if (local_user() == $item['uid']) {
-		$matches = [];
-		$first = true;
-		$cnt = preg_match_all('/\[(.*?)\]/', $item['file'], $matches, PREG_SET_ORDER);
-		if ($cnt) {
-			foreach ($matches as $mtch) {
-				$folders[] = [
-					'name' => XML::escape(FileTag::decode($mtch[1])),
-					'url' =>  "#",
-					'removeurl' => ((local_user() == $item['uid']) ? 'filerm/' . $item['id'] . '?f=&term=' . XML::escape(FileTag::decode($mtch[1])) : ""),
-					'first' => $first,
-					'last' => false
-				];
-				$first = false;
-			}
+		foreach (FileTag::fileToArray($item['file']) as $savedFolderName) {
+			$folders[] = [
+				'name' => $savedFolderName,
+				'url' => "#",
+				'removeurl' => ((local_user() == $item['uid']) ? 'filerm/' . $item['id'] . '?f=&term=' . rawurlencode($savedFolderName) : ""),
```

Comment:
Superfluous `f=&`

Reply:
It may be superfluous, but it's about `/filerm` so it feels a little out of scope of this PR. What do you think?

Reply:
I mentioned it, because you already deleted these parameter in a lot of other cases, so I thought you just forgot it ;-). But since it is a PR for RC, we could be on the save side and leave it as is.

Reply:
I only removed it for `/network` where I had to fix the logic when it was absent (even though it wasn't doing anything?). It probably is fine for all other modules but RC = not taking chances. Same reasoning for the `defaults()` to `??` migration.